src/modules/hivemq/hivemq-container.test.ts

```diff
@@ -0,0 +1,28 @@
+import { HiveMQContainer } from "./hivemq-container";
+import * as mqtt from "mqtt";
+
+describe("HiveMQContainer", () => {
+  jest.setTimeout(240_000);
+
+  // connect {
+  it("should connect to HiveMQ Community Edition via MQTT.js", async () => {
+    const hiveMQContainer = await new HiveMQContainer().start();
+
+    const testMqttClient = mqtt.connect(hiveMQContainer.getConnectionString());
+
+    testMqttClient.on("message", (topic, message) => {
```

Comment:
This test will always pass because the expectation is within a callback. You could wrap it in a promise:

```js
const message = await new Promise((resolve) => 
  testMqttClient.on("message", (topic, message) => 
    resolve(message)));

expect(message.toString()).toEqual("Test Message");

testMqttClient.end();
```

Reply:
Hope to say gotcha this time around. Had to wrap couple of definitions into functions, since strait `await new Promise` definitions looked weird in my eyes. Hope that these changes suite the codebase.

Reply:
Thanks for making the change @h2xd. 

I am nitpicking now but it seems long for a single test. Considering the mqtt library doesn't support promises, and wrapping connecting/subscribing/publishing in promises is a bit verbose, maybe the `done` callback reads better?

What do you think?

```js
it("should connect to HiveMQ Community Edition via MQTT.js", (done) => {
  new HiveMQContainer().start().then((hiveMQContainer) => {
    const testMqttClient = mqtt.connect(hiveMQContainer.getConnectionString());

    testMqttClient.on("message", (topic, message) => {
      expect(message.toString()).toEqual("Test Message");
      testMqttClient.end();
      done();
    });

    testMqttClient.on("connect", () => {
      testMqttClient.subscribe("test", (error) => {
        if (!error) {
          testMqttClient.publish("test", "Test Message");
        }
      });
    });
  });
});
```

Reply:
No problem @cristianrgreco nitpicks are just beneficial to the overall project. Will add your proposal in a second.